opencl-aot/source/utils.cpp

```diff
@@ -221,24 +227,36 @@ getOpenCLPlatform(DeviceType Type) {
         std::find(SupportedPlatformNames.begin(), SupportedPlatformNames.end(),
                   PlatformNameOnLoopIteration);
     if (Result != SupportedPlatformNames.end()) {
-      PlatformId = Platform;
-      PlatformName = PlatformNameOnLoopIteration;
-      break;
+      tie(std::ignore, ErrorMessage, CLErr) = getOpenCLDevice(Platform, Type);
+      if (!clFailed(CLErr)) {
+        PlatformId = Platform;
+        PlatformName = PlatformNameOnLoopIteration;
+        break;
+      }
     }
   }
 
-  std::string ErrorMessage;
-  if (PlatformId == nullptr) {
-    ErrorMessage += "OpenCL platform ID is empty\n";
-  }
-  if (PlatformName.empty()) {
-    ErrorMessage += "OpenCL platform name is empty\n";
+  std::string SupportedPlatforms;
+  for (const auto &Platform : DeviceTypesToSupportedPlatformNames[Type]) {
+    SupportedPlatforms += "  " + Platform + '\n';
   }
-  if (!ErrorMessage.empty()) {
-    ErrorMessage += "Failed to find any of these OpenCL platforms:\n";
-    for (const auto &SupportedPlatformName :
-         DeviceTypesToSupportedPlatformNames[Type]) {
-      ErrorMessage += "  " + SupportedPlatformName + '\n';
+  if (clFailed(CLErr)) {
+    std::map<DeviceType, std::string> DeviceTypeToDeviceTypeName{
+        {cpu, "CPU"}, {gpu, "GPU"}, {fpga_fast_emu, "accelerator"}};
```

Comment:
In case of FPGA Fast Emu, user requested ```--device=fpga_fast_emu```, so it may be unclear for him what ```accelerator``` is. So it should be renamed to something like```FPGA Fast Emu```